app/components/file_upload_preview/file_upload_item/file_upload_item.js

```diff
@@ -43,12 +43,12 @@ export default class FileUploadItem extends PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        const {file} = this.props;
-        const {file: nextFile} = nextProps;
+    componentDidUpdate(prevProps) {
+        const {file} = prevProps;
+        const {file: nextFile} = this.props;
 
         if (file.failed !== nextFile.failed && nextFile.loading) {
-            this.downloadAndUploadFile(file);
```

Comment:
Here downloadAndUploadFile was used on the previous file. So. for the first file
 `this.downloadAndUploadFile` method will be called twice - from the constructor and this componentWillReceiveProps.
Was this correct behavior?  I changed it to pass the next file instead. Is this correct?

Reply:
This is correct, but would you mind renaming the variables from `file` and `nextFile` to `prevFile` and `file` respecitvely? It'll help this be consistent with other places where we use `componentDidUpdate`